### PR TITLE
sync interface to latest PR status on deepspeed

### PR DESCRIPTION
**Pull request by delock**
This PR make idex works with latest accelerator interface on DeepSpeed (https://github.com/microsoft/DeepSpeed/pull/2221)

### CODE REVIEW

File: intel_extension_for_deepspeed/op_builder/async_io.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    def sources(self):
+        return [
+            'csrc/aio/py_lib/deepspeed_py_copy.cpp',
+            'csrc/aio/py_lib/py_ds_aio.cpp',
+            'csrc/aio/py_lib/deepspeed_py_aio.cpp',
+            'csrc/aio/py_lib/deepspeed_py_aio_handle.cpp',
+            'csrc/aio/py_lib/deepspeed_aio_thread.cpp',
+            'csrc/aio/common/deepspeed_aio_utils.cpp',
+            'csrc/aio/common/deepspeed_aio_common.cpp',
+            'csrc/aio/common/deepspeed_aio_types.cpp'
```
Comment (guoyejun) [inline intel_extension_for_deepspeed/op_builder/async_io.py:29]: all these files are referenced back to deepspeed?

**Review by rogerxfeng8 — COMMENTED** [review]
Can intel_extension_for_deepspeed.op_builder inherit the deepspeed.ops.op_builder to get the AsyncIOBuilder and SparseAttnBuilder implementations?

File: intel_extension_for_deepspeed/op_builder/async_io.py (hunk 1 above)
Comment (delock) [reply intel_extension_for_deepspeed/op_builder/async_io.py:29]: yes, these ops needs to be in the extension but kernel implementation is in deepspeed.

**Comment by delock** [comment]
> Can intel_extension_for_deepspeed.op_builder inherit the deepspeed.ops.op_builder to get the AsyncIOBuilder and SparseAttnBuilder implementations?

We can't.  The reason is deepspeed use an reflection mechanism to enumerate all_ops implemented by certain backend/extension.   So an extension needs to explicitly define the op in order to be enumerated in ALL_OPS, which is needed by build process.

**Review by guoyejun — APPROVED** [review]

**Review by rogerxfeng8 — APPROVED** [review]

**Comment by delock** [comment]
The latest downstream code can work without this patch, close this PR.  Reopen if we encounter the issue mentioned above.

**Pull request closed without merging**